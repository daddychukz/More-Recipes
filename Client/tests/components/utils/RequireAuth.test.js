import React from 'react';
import { shallow } from 'enzyme';
import Authenticate from '../../../src/utils/requireAuth';
import Resetpassword from '../../../src/components/pages/ResetPassword';

const components = Authenticate(Resetpassword);

const { WrappedComponent } = components;


let props;
let mountedComponent;

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */
const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<WrappedComponent {...props} />);
  }
  return mountedComponent;
};

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */

describe('Component: Authenticate', () => {
  beforeEach(() => {
    props = {
      isAuthenticated: true,
      history: {
        push: jest.fn()
      }
    };
    mountedComponent = undefined;
  });

  describe('Protected Routes test suite', () => {
    it('Redirects user to homepage without token', () => {
      props.isAuthenticated = false;
      expect(getComponent()).toMatchSnapshot();
    });

    it('Redirects user to homepage with invalid token', () => {
      localStorage.setItem('jwtToken', 'dfdfjjhjhjdf');
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
