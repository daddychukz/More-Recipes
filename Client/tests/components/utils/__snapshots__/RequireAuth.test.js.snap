// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Authenticate Protected Routes test suite Redirects user to homepage with invalid token 1`] = `
<Connect(ResetPassword)
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
        ],
      },
    }
  }
  isAuthenticated={true}
/>
`;

exports[`Component: Authenticate Protected Routes test suite Redirects user to homepage without token 1`] = `
<Connect(ResetPassword)
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
          Array [
            "/",
          ],
        ],
      },
    }
  }
  isAuthenticated={false}
/>
`;
